### PR TITLE
50 124 metadata store workflow manger

### PR DESCRIPTION
**Pull request by lillie-dae**
Changes required for #112.

### Description

Task Executor generating credentials for plugins.

### Status
**In Progress**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [ ] Code coverage above 70%
- [ ] TODO items resolved

### CODE REVIEW

**Comment by lillie-dae** [comment]
There are still TODO items awaiting @JJFinnerty for more information on part 2 to complete

File: src/WorkflowExecuter/Common/ConditionalParameterParser.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            }
+            return valuePairs;
+        }
+
+        private (string? Result, ParameterContext Context) ResolveMatch(string originalValue)
+        {
+            Guard.Against.NullOrWhiteSpace(originalValue);
+
+            originalValue = originalValue.Substring(2, originalValue.Length - 4).Trim();
+            //"{{ context.executions.task['2dbd1af7-b699-4467-8e99-05a0c22422b4'].'Fred' }}"
```
Comment (jackschofield23) [inline src/WorkflowExecuter/Common/ConditionalParameterParser.cs:102]: comments left in

File: tests/UnitTests/WorkflowExecuter.Tests/Services/WorkflowExecuterServiceTests.cs [hunk 2]
```diff
@@ -648,6 +657,9 @@ public async Task ProcessTaskUpdate_ValidTaskUpdateEventOneTaskDestinationDispat
         {
             var workflowInstanceId = Guid.NewGuid().ToString();
 
+            var metadata = new Dictionary<string, object>();
+            metadata.Add("a", "b");
```
Comment (jackschofield23) [inline tests/UnitTests/WorkflowExecuter.Tests/Services/WorkflowExecuterServiceTests.cs:661]: this meant to be added?

File: src/WorkflowExecuter/Common/ConditionalParameterParser.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                valuePairs.Add(match, ResolveMatch(match.Value));
+            }
+            return valuePairs;
+        }
+
+        private (string? Result, ParameterContext Context) ResolveMatch(string originalValue)
+        {
+            Guard.Against.NullOrWhiteSpace(originalValue);
+
+            originalValue = originalValue.Substring(2, originalValue.Length - 4).Trim();
```
Comment (mocsharp) [inline src/WorkflowExecuter/Common/ConditionalParameterParser.cs:101]: will `originalValue` cause an exception here?

**Review by mocsharp — APPROVED** [review]

**Comment by lillie-dae** [comment]
Going to close this PR as the PR  #185 for #114 is an continuation of this one

**Pull request closed without merging**